src/engine.ts

```diff
@@ -508,6 +556,10 @@ export function initializePolyfill() {
           }
 
           for (const child of node.childNodes) {
+            const element = child as Element;
+            if (element?.shadowRoot) {
```

Comment:
`shadowRoot` won't be available for closed shadow roots. Instead, we should add a `CQ_SHADOW` symbol that we can use to associate the instance with the element (similar to how we associate instances with nodes, and descriptors with stylesheets).